datalad_revolution/metadata/extractors/custom.py

```diff
@@ -0,0 +1,59 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Metadata extractor for custom (JSON-LD) metadata contained in a dataset"""
+
+from datalad.metadata.extractors.base import BaseMetadataExtractor
+
+import logging
+lgr = logging.getLogger('datalad.metadata.extractors.custom')
+from datalad.utils import (
+    assure_list,
+)
+
+from datalad.support.json_py import load as jsonload
+# TODO test the faith of this one
+from datalad.metadata.definitions import version as vocabulary_version
+
+from ... import utils as ut
+
+
+class MetadataExtractor(BaseMetadataExtractor):
+    def _get_dataset_metadata(self):
+        # which files to look at
+        cfg_srcfiles = self.ds.config.obtain(
+            'datalad.metadata.custom-dataset-source',
```

Comment:
don't really like the name/composition of this variable, but I cannot think of something better